### DISCUSSION TITLE
Easy way to change text on buttons?

### DISCUSSION THREAD

**Discussion by d-bucur** [bevyengine/bevy#6826] [Q&A]
I am trying to get some buttons to change their text on click. I have some entities with ButtonBundle each with a single child containing TextBundle.

This is the thing I came up. Using a query on children to find the text component. But it looks horrible tbh for such a simple thing. And it's a mess to refactor into a separate function.

Is there a more sane way to do this?

```rust
fn player_input(
    mut interaction_query: Query<
        (&Interaction, &mut BackgroundColor, &GridPosition, &Children),
        (Changed<Interaction>, With<Button>),
    >,
    mut children_query: Query<&mut Text>,
    mut grid: ResMut<Grid>,
) {
    for (interaction, mut color, grid_position, children) in &mut interaction_query {
        match *interaction {
            Interaction::Clicked => {
                for child in children {
                    for mut text in children_query.get_mut(*child) {
                        text.sections[0].value = "T".into();
                    }
                }
...
```

**Comment by ThierryBerger** [comment]
I would add a component storing the text entity to your `Interaction` entity:

```rs
struct ChangeTextOnClicked
{
    text_entity: Entity,
}
```

Then you can replace your children_query iteration by a simpler direct get for the Text component on the `text_entity`.

This approach can also help with having different `Interaction` behaviours (buttons not changing text ?).
